homeassistant/components/tts/__init__.py

```diff
@@ -910,11 +924,15 @@ def async_remove_from_mem(_: datetime) -> None:
             ),
         )
 
-    async def async_read_tts(self, filename: str) -> tuple[str | None, bytes]:
+    async def async_read_tts(self, token: str) -> tuple[str | None, bytes]:
         """Read a voice file and return binary.
 
         This method is a coroutine.
         """
+        filename = self.filename_token.get(token)
+        if not filename:
+            raise HomeAssistantError(f"{token} was not recognized!")
```

Comment:
This should raise a 401, as we shouldn't expose if the key existed or not.

Reply:
Wouldn't raising a different error here than in the other conditions (404) make it obvious that the key doesn't exist?